spec/recurly/plan_spec.rb

```diff
@@ -41,6 +44,54 @@
 </unit_amount_in_cents>\
 </plan>
 XML
+    end
+  end
+
+  describe 'when pricing_model is ramp' do
```

Comment:
Is this really the only sort of testing done for the ruby lib? Just serialization from object->xml but not the other way around? And nothing dealing with the various requests?  🤔

Reply:
Yeah, it's interesting. Some of the tests (especially those in `subscription_spec.rb`) do go into more detail, making stubbed requests and whatnot, but that feels a bit nonsensical to me.

I like the idea of the specs being as lean as the source file - the `Plan` class doesn't do much on its own, it merely wires things up for use by the `Resource` implementation (which itself is concerned with making http requests, etc.).

Reply:
Yea, I dunno, just doesn't personally give me a ton of confidence on the end-to-end bit of it. It just feels weird not to test every code path to ensure that bit of confidence. Like, it can serialize, but is it set up right to ingest xml? Does it work for initially ingested XML and updating based on a PUT update? Just...feels weird.  😅

Reply:
> Like, it can serialize, but is it set up right to ingest xml?

That's basically the responsibility from `Resource#from_xml`, which has its own tests.

> Does it work for initially ingested XML and updating based on a PUT update?

That's the responsibility of `Resource#to_xml` (serializing the resource instance into an xml payload) and `Resource#save`, which also have their own tests.